### PR TITLE
UIDATAIMP-35: Create reusable react Drag and Drop component

### PR DESCRIPTION
**Pull request by Nick-Yenin**

### CODE REVIEW

File: src/components/ImportJobs/components/FileUpload/FileUpload.css [hunk 1]
```diff
@@ -16,3 +16,14 @@
   margin-bottom: 0.5rem;
   font-weight: 600;
 }
+
+
+.activeUpload {
+  background: black;
```
[the remark this answers is not in the document]
Comment (VictorSoroka1) [reply src/components/ImportJobs/components/FileUpload/FileUpload.css]: For colors use variables from `@import "@folio/stripes-components/lib/variables.css";`. If needed ask Filip designer for the exact one. Check other places.

File: src/components/ImportJobs/components/FileUpload/FileUpload.css [hunk 2]
```diff
@@ -16,3 +16,14 @@
...
+  color: #fff;
+  border-style: dashed;
+}
+
+.activeUploadTitle {
+  font-size: 30px;
```
Comment (VictorSoroka1) [inline src/components/ImportJobs/components/FileUpload/FileUpload.css]: For font-size use `rem` unit. Firstly check whether such variable exists in variables file.

File: src/components/ImportJobs/ImportJobs.js [hunk 3]
```diff
@@ -2,13 +2,52 @@ import React from 'react';
 import { FormattedMessage } from 'react-intl';
 
 import FileUpload from './components/FileUpload';
+import css from './components/FileUpload/FileUpload.css';
```
Comment (VictorSoroka1) [inline src/components/ImportJobs/ImportJobs.js]: Please, add empty space before styles import. Check other places.

File: src/components/ImportJobs/components/FileUpload/FileUpload.js [hunk 4]
```diff
@@ -1,35 +1,125 @@
...
-          buttonStyle="primary"
-          onClick={noop}
-        >
-          {uploadBtnText}
-        </Button>
-      </div>
-    );
-  }
-}
+
```
Comment (VictorSoroka1) [inline src/components/ImportJobs/components/FileUpload/FileUpload.js]: Remove excess empty line.

File: src/components/ImportJobs/components/FileUpload/FileUpload.js [hunk 5]
```diff
@@ -1,35 +1,125 @@
...
+      rejectClassName={rejectClassName}
+      disabledClassName={disabledClassName}
+      maxSize={maxSize}
+      getDataTransferItems={getDataTransferItems}
+    >
+      {({ open }) => (
+        <React.Fragment>
+          {children}
+          <span className={titleClassName}>{title}</span>
+          {!isDropZoneActive &&
```
Comment (VictorSoroka1) [inline src/components/ImportJobs/components/FileUpload/FileUpload.js]: Is this confirmed behavior to hide button if d&d way is used?

File: src/components/ImportJobs/components/FileUpload/FileUpload.js [hunk 6]
```diff
@@ -1,35 +1,125 @@
...
+};
+
+const getUsedStyle = (styleFromProps, classNameFromProps, initialStyle) => {
+  if (styleFromProps) return styleFromProps;
+  if (classNameFromProps) return null;
+
+  return initialStyle;
+};
+
+const initStyle = {
```
Comment (VictorSoroka1) [inline src/components/ImportJobs/components/FileUpload/FileUpload.js]: Can we have this as initClass? I just do not want to have styles block defined not inside `css` file.

File: src/components/ImportJobs/components/FileUpload/FileUpload.js [hunk 7]
```diff
@@ -1,35 +1,116 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import noop from 'lodash/noop';
+import ReactDropzone from 'react-dropzone';
 
 import { Button } from '@folio/stripes/components';
 
+import classNames from 'classnames/bind';
```
Comment (VictorSoroka1) [inline src/components/ImportJobs/components/FileUpload/FileUpload.js]: Please move classNames right after `ReactDropzone `.

File: src/components/ImportJobs/ImportJobs.js [hunk 8]
```diff
@@ -1,17 +1,65 @@
...
+
+  getMessageById = (idEnding, moduleName = 'ui-data-import') => {
+    const id = `${moduleName}.${idEnding}`;
+
+    return this.props.intl.formatMessage({ id });
+  };
+
   render() {
+    const titleMessageIdEnding = this.state.isDropZoneActive ? 'activeUploadTitle' : 'uploadTitle';
+
```
Comment (VictorSoroka1) [inline src/components/ImportJobs/ImportJobs.js]: Remove excess empty line.

File: src/components/ImportJobs/components/FileUpload/initStyle.css [hunk 9]
```diff
@@ -0,0 +1,9 @@
+.defaultFileUploader {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  flex-direction: column;
+  text-align: center;
+  border: 1px solid black;
+  height: 200px;
```
Comment (VictorSoroka1) [inline src/components/ImportJobs/components/FileUpload/initStyle.css]: Let's have it as `min-height`.

**Comment by oleksii-popov** [comment]
Approved from my side.

**Pull request merged**